pkg/sql/plan/function/builtin/unary/curdate.go

```diff
@@ -0,0 +1,82 @@
+// Copyright 2022 Matrix Origin
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package unary
+
+import (
+	"github.com/matrixorigin/matrixone/pkg/common/moerr"
+	"github.com/matrixorigin/matrixone/pkg/container/types"
+	"github.com/matrixorigin/matrixone/pkg/container/vector"
+	"github.com/matrixorigin/matrixone/pkg/vm/process"
+	"time"
+)
+
+const dayFromUnixEpoch = 719163
+
+func CurDate(vec []*vector.Vector, proc *process.Process) (*vector.Vector, error) {
+	if len(vec) == 0 {
+		return curDate(proc)
+	}
+	return curDateWithArg(vec[0], proc)
+}
+
+func curDateWithArg(vec *vector.Vector, proc *process.Process) (resultVec *vector.Vector, err error) {
+	defer func() {
+		if err != nil && resultVec != nil {
+			resultVec.Free(proc.Mp())
+		}
+	}()
+	if !vec.IsScalar() {
+		err = moerr.NewNotSupported("curdate with non-scalar argument")
```

Comment:
why we do not support non-scalar?

Reply:
because this `withArgs function` is a hack which is used to replace  `curdate()+$arg`  with `curdate($arg)` in plan, to return days from unixEpoch. it may not need support non-scalar.

Reply:
seems select curdate() + intcol from t is a perfectly reasonable use.   why $arg is special?

Reply:
actually why do you bother?   if date can add an integer, let the add function handle this.